tests/integration/test_documentation_files_validation.py

```
@@ -0,0 +1,357 @@
+
+Tests ensure that:
+- All markdown files are valid and parseable
+- Links within markdown files are correctly formatted
+- Code blocks have proper language identifiers
+- Tables are properly formatted
+- Headings follow a logical hierarchy
+"""
+import os
+import re
+from pathlib import Path
+from typing import List, Tuple
```

Ruff (0.14.8)

`typing.List` is deprecated, use `list` instead

(UP035)